docs/ldml/tr35-keyboards.md

```diff
@@ -308,6 +308,263 @@ For extensibility, the `<special>` element will be allowed at nearly every level
 
 See [Element special](tr35.md#special) in Part 1.
 
+## Normalization
+
+Normalization will not typically be the responsibility of the keyboard author, rather this will be managed by the keyboard implementation.
+The keyboard implementation will apply normalization as appropriate when matching transform rules and `<display>` value matching.
+Output from the keyboard, following application of all transform rules, will be normalized to implementation- or application-requested form.
```

Comment:
(also point 4. below) How does application request normalization form? This seems to be out of scope for the spec, and out of scope for input method, if the application needs NFD, it should perform NFD itself.

Reply:
That's up to the implementation.

The idea here is that keyboard implementation could be a 'platform' vs the end 'application' running on top of that platform.

An implementation might always want to output in NFC, that might be a common situation.

Or, perhaps applications on top of that implementation can request NFC or NFD.

Reply:
@macchiati also had pointed out that we should define the term "implementation" more specifically. I will do the same for "application".

Reply:
What do you think of this:

```suggestion
Output from the keyboard, following application of all transform rules, will be normalized to the appropriate form for the keyboard implementation.
```

Reply:
This looks good.